### ISSUE TITLE
After using animation, route to other pages, click the button, and the View will shake（I don't know where the problem is router, navigation or animated)

### ISSUE THREAD

**Issue by laohan** [expo/expo#33866]
### Minimal reproducible example

https://github.com/laohan/animated-flash/tree/main

### Which package manager are you using? (Yarn is recommended)

yarn

### If the issue is web-related, please select the bundler (`web.bundler` in the `app.json`)

None

### Summary

When navigating from page A containing animation to page B and clicking a button on page B, the View will flicker, thus triggering layout frequently. This phenomenon only occurs on iOS.

https://github.com/user-attachments/assets/3b817b96-d7e6-4275-91b5-a261ac3ce817

### Environment

```text
expo-env-info 1.2.1 environment info:
    System:
      OS: macOS 15.2
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 23.1.0 - /opt/homebrew/bin/node
      Yarn: 1.22.22 - /opt/homebrew/bin/yarn
      npm: 10.8.2 - /opt/homebrew/bin/npm
      Watchman: 2024.11.04.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.16.2 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 24.2, iOS 18.2, macOS 15.2, tvOS 18.2, visionOS 2.2, watchOS 11.2
    IDEs:
      Android Studio: 2024.2 AI-242.23339.11.2421.12550806
      Xcode: 16.2/16C5032a - /usr/bin/xcodebuild
    npmPackages:
      expo: ~52.0.23 => 52.0.23 
      expo-router: ~4.0.15 => 4.0.15 
      react: 18.3.1 => 18.3.1 
      react-dom: 18.3.1 => 18.3.1 
      react-native: 0.76.5 => 0.76.5 
      react-native-web: ~0.19.13 => 0.19.13 
    npmGlobalPackages:
      eas-cli: 13.4.2
    Expo Workflow: bare
```

**Comment by laohan** [comment]
There is no problem with router.replace navigating to the page

**Comment by MatLish00010** [comment]
Same problem , any updates?

**Comment by chrfalch** [comment]
This is probably related to the use of the new architecture in the project (see `app.json` and the `newArchEnabled` field) and is caused by some issues with navigation animations under the new architecture - could you try to verify this by setting the `newArchEnabled` field to `false`, run prebuild and then build/run?

**Comment by MatLish00010** [comment]
Yes, but do you know when expo will fix it? Because this is important i think

**Comment by marklawlor** [comment]
Hey! Thanks for opening the issue. Seems that this issue is related to `react-navigation` library which is a dependency of Expo Router. Can you also post your issue in [this repo](https://github.com/react-navigation/react-navigation) so that it's notified to the maintainers of that library? This will help us fix the issue faster since it's upto the maintainers of that library to investigate it.

**Comment by laohan** [comment]
> Hey! Thanks for opening the issue. Seems that this issue is related to `react-navigation` library which is a dependency of Expo Router. Can you also post your issue in [this repo](https://github.com/react-navigation/react-navigation) so that it's notified to the maintainers of that library? This will help us fix the issue faster since it's upto the maintainers of that library to investigate it.

Thanks, I've submitted the[ issue](https://github.com/react-navigation/react-navigation/issues/12383) elsewhere.

**Comment by Tieu-Mai-Nhi** [comment]
same problem, I have a part of the bottom tab that keeps flashing

**Comment by iiagodias** [comment]
> same problem, I have a part of the bottom tab that keeps flashing

here too

**Comment by RainPlays09** [comment]
> same problem, I have a part of the bottom tab that keeps flashing

Same problem here

**Comment by RainPlays09** [comment]
> > Hey! Thanks for opening the issue. Seems that this issue is related to `react-navigation` library which is a dependency of Expo Router. Can you also post your issue in [this repo](https://github.com/react-navigation/react-navigation) so that it's notified to the maintainers of that library? This will help us fix the issue faster since it's upto the maintainers of that library to investigate it.
> 
> Thanks, I've submitted the[ issue](https://github.com/react-navigation/react-navigation/issues/12383) elsewhere.

In your repo, someone[ commented](https://github.com/react-navigation/react-navigation/issues/12383#issuecomment-2646637753) that the issue is with the Header component, due to which the onLayout height keeps recalculating. I tried to use the workaround but it doesn't work. But I tried to create a Tab Custom Navigator (not Stack), the issue was fixed so I think you should create your custom Navigator to fix the issue

**Comment by laohan** [comment]
> > > 嘿！感谢您提出这个问题。这个问题似乎与`react-navigation`Expo Router 的依赖库有关。您也可以在[此 repo](https://github.com/react-navigation/react-navigation)中发布您的问题，以便通知该库的维护者吗？这将帮助我们更快地解决问题，因为这取决于该库的维护者来调查它。
> > 
> > 
> > 谢谢，我已将该[问题](https://github.com/react-navigation/react-navigation/issues/12383)提交到其他地方。
> 
> 在您的问题中，有人[评论](https://github.com/react-navigation/react-navigation/issues/12383#issuecomment-2646637753)说问题出在 Header 组件上，因此 onLayout 高度不断重新计算。我尝试使用解决方法，但不起作用。但我尝试创建一个 Tab 自定义导航器（不是 Stack），问题已修复，所以我认为您应该创建自定义导航器来修复该问题

Thanks!
